src/main/java/sana/AddCommand.java

```diff
@@ -0,0 +1,136 @@
+package sana;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a command for adding tasks to the task list.
+ */
+public class AddCommand extends Command {
+
+    /**
+     * Constructs an AddCommand object.
+     *
+     * @param cmd       The command type (e.g., "todo", "deadline", "event").
+     * @param arguments The arguments associated with the command.
+     */
+    public AddCommand(String cmd, String arguments) {
+        super(cmd, arguments);
+    }
+
+    /**
+     * Executes the add command by adding a task to the task list and saving to file.
+     *
+     * @param tasks   The TaskList object containing the list of tasks.
+     * @param ui      The Ui object for handling user interface.
+     * @param storage The Storage object for saving and loading tasks.
+     * @throws SanaException If an error specific to Sana occurs during execution.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws SanaException {
+        String cmd = getCmd();
+        String arguments = getArguments();
+        switch (cmd) {
+        case "todo":
+            if (arguments.isBlank()) {
+                throw new SanaException("OOPS!!! The description of a todo cannot be empty.");
+            } else {
+                Task newTodo = new Todo(arguments, false);
+                tasks.add(newTodo);
+                storage.save("/Users/ariellacallista/Desktop",
+                        "/Users/ariellacallista/Desktop/SanaTasks.txt", newTodo);
+
+                System.out.println("Got it. I've added this task:\n" + newTodo + "\n"
```

Comment:
`System.out.println("Got it. I've added this task:\n" + newTodo + "\n"`

Based on your class structure, all of the outputs (eg. `system.out.println()`) should be in the Ui class to make it easier to change to a GUI. You can instead add a `showTaskAddedMessage()` method to the Ui class and call it here. Similar for all of the other types of tasks.